pontoon/base/migrations/0131_translation_active_populate.py

```diff
@@ -0,0 +1,68 @@
+# -*- coding: utf-8 -*-
+# Generated by Django 1.11.15 on 2018-08-30 12:36
+from __future__ import unicode_literals
+
+import datetime
+
+from django.db import migrations
+
+
+def set_active_translations(apps, schema_editor):
+    Translation = apps.get_model('base', 'Translation')
+
+    # All approved translations are active.
+    Translation.objects.filter(approved=True).update(active=True)
+    print(datetime.datetime.time(datetime.datetime.now()))
+
+    # All fuzzy translations are active.
+    # Note that this wasn't the case until this patch - if a fuzzy translation
+    # had a newer sibling suggestion, that suggestion was active.
+    # However, this is a bug, and we're fixing it here.
+    Translation.objects.filter(fuzzy=True).update(active=True)
+    print(datetime.datetime.time(datetime.datetime.now()))
+
+    # The most recent unreviewed suggestion for any given combination of
+    # (locale, entity, plural_form) is active, unless it has approved or fuzzy
+    # siblings.
+    # Note that currenty this logic applies to all suggestions, even rejected.
+    # However, this is a bug, and we're fixing it here.
+    unreviewed_pks = set()
+    candidates = Translation.objects.filter(
+        approved=False,
+        fuzzy=False,
+        rejected=False,
+    )
+    print(datetime.datetime.time(datetime.datetime.now()))
+
+    for candidate in candidates:
+        siblings = Translation.objects.filter(
+            entity=candidate.entity,
+            locale=candidate.locale,
+            plural_form=candidate.plural_form,
+        )
+        if not siblings.filter(active=True).exists():
+            most_recent = siblings.latest('date')
+            unreviewed_pks.add(most_recent.pk)
```

Comment:
I think this block can be optimized. Right now it does a lot of computation several times (if an entity has several suggestions, then this logic is applied to all suggestions). However, given how complex this model is, finding the optimal solution is going to be quite complex, and my brain is definitely not able to do that on a Thursday evening. :P

Reply:
True. I thought about that for 5 minutes and then gave up and tried to measure it first. Check out the [timings](https://github.com/mozilla/pontoon/pull/1072#issue-212123351).

This part takes 8 minutes, which means that even if we get it down to a minute, there's still 75% of the migration time left. We'll likely need to take Pontoon down briefly anyhow.